File: simple-anal/anal.js
var r2pipe = require('r2pipe');


if (!process.argv[2]) {
    console.error('You must supply a file to analize');
    process.exit(1);
}

var file = process.argv[2];


var r2 = r2pipe.pipeSync(file)
console.log('===================================');
console.log('              IMPORTS');
console.log('===================================');
console.log(r2.syscmdj('rabin2 -j -i ' + file));

console.log('===================================');
console.log('              STRINGS');
console.log('===================================');
console.log(r2.syscmdj('rabin2 -j -zz ' + file));

console.log('===================================');
console.log('              SECTIONS');
console.log('===================================');
console.log(r2.syscmdj('rabin2 -j -S ' + file));

console.log('===================================');
console.log('              SYMBOLS');
console.log('===================================');
console.log(r2.syscmdj('rabin2 -j -s ' + file));

r2.quit();

File: simple-anal/anal.py
import sys
import r2pipe

r2 = r2pipe.open("-")

print "============================"
print "        IMPORTS"
print "============================"
print r2.syscmdj("rabin2 -j -i %s" % sys.argv[1])

print "============================"
print "        STRINGS"
print "============================"
print r2.syscmdj("rabin2 -j -zz %s" % sys.argv[1])

print "============================"
print "        SECTIONS"
print "============================"
print r2.syscmdj("rabin2 -j -S %s" % sys.argv[1])

print "============================"
print "        SYMBOLS"
print "============================"
print r2.syscmdj("rabin2 -j -s %s" % sys.argv[1])
